common/include/bmp180_protocol.h

```diff
@@ -0,0 +1,44 @@
+#ifndef BMP180_PROTOCOL_H_
+#define BMP180_PROTOCOL_H_
+
+#include "abstract_protocol.h"
+
+class Bmp180Protocol: public DummyProtocol {
+public:
+	Bmp180Protocol() {}
+	~Bmp180Protocol() {}
+
+	virtual error_condition UnpackFrame(
+			const vector<uint8_t>& buffer, vector<uint8_t>& unpacked_data) {
+		if (ExtractValidData(buffer, unpacked_data) != true) {
+			return make_error_condition(errc::bad_message);
+		}
+		return kNoError;
+	};
+protected:
+	virtual error_condition CalcCheckCode(\
+			const vector<uint8_t>& buffer, vector<uint8_t> check_code) {
+		if ((buffer.back() != check_code.back()) || (buffer.at(buffer.size() - 2) \
+				!= check_code.at(check_code.size() - 2))) {
+			return make_error_condition(errc::bad_message);
+		}
+		return kNoError;
+	};
+	virtual bool ExtractValidData(const vector<uint8_t>& buffer,\
+			vector<uint8_t>& unpacked_data) {
+		if (IsFrameComplete(buffer) != true) {
```

Comment:
In C++ the type of if condition statement is `bool`, so we need to avoid something like:
```
int status = 0;
...
if (status) {
...
}
```
Needs to be:
```
if (status == 0) {
...
}
```
But if the returning value type is already `bool`, no need for a comparison:
```suggestion
		if (IsFrameComplete(buffer)) {
```